bucket4j/src/main/java/com/linecorp/armeria/server/throttling/bucket4j/TokenBucketSpec.java

```diff
@@ -0,0 +1,184 @@
+        if (options.containsKey(OVERDRAFT)) {
+            final long overdraftLimit = Long.parseLong(options.get(OVERDRAFT));
+            if (options.containsKey(INITIAL)) {
+                final long initialSize = Long.parseLong(options.get(INITIAL));
+                return BandwidthLimit.of(limit, overdraftLimit, initialSize, period);
+            } else {
+                return BandwidthLimit.of(limit, overdraftLimit, period);
+            }
+        } else {
+            return BandwidthLimit.of(limit, period);
+        }
+    }
+
+    /**
+     * Creates a new {@link TokenBucket} from a comma-separated {@code specification} string
+     * that conforms to the following format:
+     * <pre>{@code
+     * <bandwidth limit 1>[, <bandwidth limit 2>[, etc.]]
+     * }</pre>
+     * The order of elements inside {@code specification} is not defined.
+     * For example:
+     * <ul>
+     *   <li>{@code 100;window=60;burst=1000, 50000;window=3600}</li>
+     * </ul>
+     *
+     * @param specification the specification used to create a {@link BandwidthLimit}.
+     *                      Empty {@link String} permitted to specify empty {@link TokenBucket}.
+     */
+    static TokenBucket parseTokenBucket(String specification) {
+        requireNonNull(specification, "specification");
+        final TokenBucketBuilder builder = TokenBucket.builder();
+        if (specification.isEmpty()) {
+            // empty specification allowed here
+            return builder.limits().build();
+        }
+        final List<BandwidthLimit> limits = new ArrayList<>(2);
+        for (String limitSpec: LIMITS_SPLITTER.split(specification)) {
+            limits.add(parseBandwidthLimit(limitSpec));
+        }
+        return builder.limits(limits).build();
+    }
+
+    /**
+     * Returns a string representation of the {@link BandwidthLimit} in the following format:
+     * <pre>{@code
+     * <limit>;window=<period(in seconds)>[;burst=<overdraftLimit>][;policy="token bucket"]
+     * }</pre>
+     * For example: "100;window=60;burst=1000".
+     *
+     * @return A {@link String} representation of the {@link BandwidthLimit}.
+     */
+    @Nullable
+    static String toString(@Nullable BandwidthLimit bandwidthLimit) {
+        if (bandwidthLimit == null) {
+            return null;
+        }
+        final long limit = bandwidthLimit.limit();
+        final StringBuilder sb = new StringBuilder().append(limit);
+        sb.append(OPTIONS_SEPARATOR).append(PERIOD).append(KEY_VALUE_SEPARATOR)
+          .append(bandwidthLimit.period().getSeconds());
+        final long overdraftLimit = bandwidthLimit.overdraftLimit();
+        if (overdraftLimit > limit) {
+            sb.append(OPTIONS_SEPARATOR).append(OVERDRAFT).append(KEY_VALUE_SEPARATOR).append(overdraftLimit);
+        }
+        //sb.append(OPTIONS_SEPARATOR).append(INITIAL).append(KEY_VALUE_SEPARATOR)
+        //  .append(bandwidthLimit.initialSize());
```

Comment:
Aren't we going to add initial size?

Reply:
This is intentional, but unfortunately asymmetric. The initial size of the limit is important as a configuration parameter, which this spec is parsing. But the spec is also used to communicate parameters of the quota policy to the client as part of `RateLimit-Limit` header. The initial size of the limit is not important as part of the [quota policy](https://tools.ietf.org/id/draft-polli-ratelimit-headers-00.html), since it's only applicable at the warmup.
Let me know if this is too confusing. I could add more explanation of that into the comment or javadoc.

Reply:
Thanks for the explanation! Then, let's just remove this code and add the comment on why we didn't do this for the initial size. 😉

Reply:
Gentle ping - Good to go once this comment is addressed.

Reply:
Sorry for the delay. Just pushed the fix.